code/modules/clothing/spacesuits/rig/modules/utility.dm

```diff
@@ -380,4 +377,102 @@
 	jets.holder = null
 	jets.trail.set_up(jets)
 
-/obj/item/rig_module/foam_sprayer
+/obj/item/rig_module/autodoc
+	name = "mounted autodoc"
+	desc = "A complex surgery system for almost all your needs. Can't treat organ damage, but capable to fix internal bleeding."
+	icon_state = "autodoc"
+	toggleable = 1
+	disruptive = 1
+
+	active_power_cost = 50
+
+	activate_string = "Activate autodoc"
+	deactivate_string = "Deactivate autodoc"
+
+	interface_name = "Autodoc"
+	interface_desc = "Module with set of instruments that is capable to preform basic surgery on user"
+
+	var/list/surgery_list = new()
+
+#define DR_RIG_BRUTE 0x1
+#define DR_RIG_BURN 0x2
+#define DR_RIG_BROKEN 0x4
+#define DR_RIG_IB 0x8
+/datum/rig_surgery
+	var/surgery_type = 0
+	var/obj/item/organ/external/ex_organ = null
+
+/obj/item/rig_module/autodoc/activate()
+	. = ..()
+	if(!.)
+		return
+	to_chat(holder.wearer, SPAN_NOTICE("With quiet hum internal machinery starts checking your body..."))
+	var/mob/living/carbon/human/H = holder.wearer
+
+	for(var/obj/item/organ/external/E in H.bad_external_organs)
+		to_chat(holder.wearer, "Found wounded organ: "+SPAN_WARNING("[E]"))
+		var/datum/rig_surgery/patchnote = new()
+		for(var/datum/wound/internal_bleeding/IB in E.wounds)
+			patchnote.surgery_type += DR_RIG_IB
+			to_chat(holder.wearer, SPAN_WARNING("| Internal bleeding"))
+			break
+		if (E.status & ORGAN_BROKEN)
+			patchnote.surgery_type += DR_RIG_BROKEN
+			to_chat(holder.wearer, SPAN_WARNING("| Bone fractures"))
+		if (E.brute_dam)
+			patchnote.surgery_type += DR_RIG_BRUTE
+			to_chat(holder.wearer, SPAN_WARNING("| Brute damage"))
+		if (E.burn_dam)
+			patchnote.surgery_type += DR_RIG_BURN
+			to_chat(holder.wearer, SPAN_WARNING("| Burn damage"))
+		if(patchnote.surgery_type)
+			patchnote.ex_organ = E
+			surgery_list.Add(patchnote)
+	if(!surgery_list.len)
+		to_chat(holder.wearer, SPAN_WARNING("No brute, burn damage, internal bleeding or fracture was found."))
+		deactivate()
+
+/obj/item/rig_module/autodoc/deactivate()
+	. = ..()
+	if(!.)
+		return
+	surgery_list.Cut()
+
+/obj/item/rig_module/autodoc/Process()
+	. = ..()
+	for(var/datum/rig_surgery/patchnote in surgery_list)
+		var/obj/item/organ/external/ex_organ = patchnote.ex_organ
+		if(!ex_organ)
+			CRASH()
```

Comment:
How about `surgery_list -= patchnote` to prevent continuous crashing?

Reply:
Well, it was late night brainfart about gibbed organs in action, it honestly should just remove patchnote at all.